lms/djangoapps/django_comment_client/forum/tests.py

```diff
@@ -1312,6 +1312,15 @@ def _test_unicode_data(self, text, mock_request):
         self.assertEqual(response_data["discussion_data"][0]["title"], text)
         self.assertEqual(response_data["discussion_data"][0]["body"], text)
 
+    def _test_invalid_course_id(self):
+        """ Asserts that Http404 is raised when the course id is not valid. """
+        request = RequestFactory().get("dummy_url")
```

Comment:
Same here. 

I think , we can create a common class and let these test inherit. You can pass on the specific view method and args that you want to the method to call with.